### ISSUE TITLE
Analyze @namespace jsdoc annotations

### ISSUE THREAD

**Issue by justinfagnani** [Polymer/polymer-analyzer#481]
We need to find variable declarations with a `@namespace` annotation and create ScannedNamespace and Namespace features.

Examples:

```javascript
/**
 * @namespace
 */
const Polymer = function() { … }

/**
 * @namespace
 * @memberof Polymer
 */
Polymer.dom = function() { … }

/**
 * @memberof Polymer.dom
 */
Polymer.dom.foo = …;

/**
 * @memberof Polymer.dom
 */
Polymer.dom.bar = …;

/**
 * @namespace
 */
Polymer.Async = {
  run() {...},
};
```

`@namespace` can take a name to the right of it, otherwise we can get the name from the left side of the assignment. 

`@memberof` should be handled as a separate issue.

**Comment by rictic** [comment]
Do we need the `@memberof`? Can we somehow notice assignments to namespaces automatically?

I guess it's easy to start explicit and then start inferring. Ok, nm, talked myself into this plan.

**Comment by justinfagnani** [comment]
🥇

**Comment by justinfagnani** [comment]
We also need a top-level `namespaces` property of elements.json, or we need to make elements.json have a nested structure.

@rictic ^ thoughts?

**Comment by rictic** [comment]
It seems like, nested or no, we need either `namespaces` or just, like, `objects`

**Comment by justinfagnani** [comment]
This was fixed by #486
